Name: Swipe to Delete Videos
Version: 1.3.5-1
Author: Jason Robitaille
Description: Adds the ability to swipe to delete videos  (MIT license)

--- .orig/usr/palm/applications/com.palm.app.videoplayer.launcher/app/controllers/browser-assistant.js
+++ /usr/palm/applications/com.palm.app.videoplayer.launcher/app/controllers/browser-assistant.js
@@ -15,6 +15,8 @@
 			itemTemplate: 'browser/browser-entry',
 			filterFunction: this.requestVideosList.bind(this),
 			optimizedOptIn: true,
+			swipeToDelete: true,
+			autoconfirmDelete: true,
 			formatters: {
 				videoPictureUrl: this.videoPictureUrlFormatter,
 				videoSize: this.videoSizeFormatter.bind(this),
@@ -57,6 +59,8 @@
 	
 	activate: function(){
 		Util.markForeground(this.controller, true);
+		this.handleDelete = this.handleDelete.bindAsEventListener(this);
+		this.controller.listen('videoslist', Mojo.Event.listDelete, this.handleDelete);
 	},
 	
 	deactivate: function(){
@@ -61,6 +65,7 @@
 	
 	deactivate: function(){
 		Util.markForeground(this.controller, false);
+		this.controller.stopListening('videoslist', Mojo.Event.listDelete, this.handleDelete);
 	},
 	
 	showNoVideosDiv: function(){		
@@ -216,6 +221,33 @@
 		}
 	},
 	
+	handleDelete: function(event){
+		event.stop(event);
+		this.deletePath = event.item.videoPath;
+		this.controller.showAlertDialog({
+			onChoose: function(value) {
+				if (value == 'Yes') {
+					this.controller.serviceRequest('palm://com.palm.mediadb/', {
+						method: 'deletefile',
+						parameters: {
+							path: this.deletePath
+						},
+						onSuccess: function() {
+							this.controller.showBanner("Video deleted!", "");
+						}.bind(this)
+					});
+				}
+			},
+			title: $L("Delete?"),
+			message: $L("Are you sure you want to delete \"" + event.item.videoNameFormatted + "\"?" + " This cannot be undone."),
+			choices:[
+				{label:$L('Yes'), value:"Yes", type:'affirmative'},  
+				{label:$L('No'), value:"No", type:'negative'} 
+			]
+		});
+		
+	},
+	
 	/**
 	 * Taken from the picker in the framework.
 	 */
--- .orig/usr/palm/applications/com.palm.app.videoplayer/app/controllers/browser-assistant.js
+++ /usr/palm/applications/com.palm.app.videoplayer/app/controllers/browser-assistant.js
@@ -15,6 +15,8 @@
 			itemTemplate: 'browser/browser-entry',
 			filterFunction: this.requestVideosList.bind(this),
 			optimizedOptIn: true,
+			swipeToDelete: true,
+			autoconfirmDelete: true,
 			formatters: {
 				videoPictureUrl: this.videoPictureUrlFormatter,
 				videoSize: this.videoSizeFormatter.bind(this),
@@ -57,6 +59,8 @@
 	
 	activate: function(){
 		Util.markForeground(this.controller, true);
+		this.handleDelete = this.handleDelete.bindAsEventListener(this);
+		this.controller.listen('videoslist', Mojo.Event.listDelete, this.handleDelete);
 	},
 	
 	deactivate: function(){
@@ -61,6 +65,7 @@
 	
 	deactivate: function(){
 		Util.markForeground(this.controller, false);
+		this.controller.stopListening('videoslist', Mojo.Event.listDelete, this.handleDelete);
 	},
 	
 	showNoVideosDiv: function(){		
@@ -216,6 +221,33 @@
 		}
 	},
 	
+	handleDelete: function(event){
+		event.stop(event);
+		this.deletePath = event.item.videoPath;
+		this.controller.showAlertDialog({
+			onChoose: function(value) {
+				if (value == 'Yes') {
+					this.controller.serviceRequest('palm://com.palm.mediadb/', {
+						method: 'deletefile',
+						parameters: {
+							path: this.deletePath
+						},
+						onSuccess: function() {
+							this.controller.showBanner("Video deleted!", "");
+						}.bind(this)
+					});
+				}
+			},
+			title: $L("Delete?"),
+			message: $L("Are you sure you want to delete \"" + event.item.videoNameFormatted + "\"?" + " This cannot be undone."),
+			choices:[
+				{label:$L('Yes'), value:"Yes", type:'affirmative'},  
+				{label:$L('No'), value:"No", type:'negative'} 
+			]
+		});
+		
+	},
+	
 	/**
 	 * Taken from the picker in the framework.
 	 */
